### PR TITLE
Bug 1276602 - fixes error when scaling dc with --timeout

### PR DESCRIPTION
**Pull request by fabianofranz**

### CODE REVIEW

**Comment by fabianofranz** [comment]
@ironcladlou PTAL

**Comment by ironcladlou** [comment]
cc @kargakis

File: pkg/deploy/scaler/scale.go [hunk 1]
```diff
@@ -55,7 +55,11 @@ func (scaler *DeploymentConfigScaler) Scale(namespace, name string, newSize uint
 		return err
 	}
 	if waitForReplicas != nil {
-		rc, err := scaler.rcClient.ReplicationControllers(namespace).Get(name)
+		dc, err := scaler.dcClient.DeploymentConfigs(namespace).Get(name)
```
Comment (ironcladlou) [inline pkg/deploy/scaler/scale.go]: Looks like that could never have worked... Let's add test coverage.

Comment (kargakis) [inline pkg/deploy/scaler/scale.go]: You definitely need to update tests ...

**Comment by fabianofranz** [comment]
fyi @smarterclayton

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
@openshift/team-project-commiters

File: pkg/deploy/scaler/scale.go (hunk 1 above)
Comment (fabianofranz) [inline pkg/deploy/scaler/scale.go]: Done

**Comment by fabianofranz** [comment]
@openshift/team-project-committers

**Comment by ironcladlou** [comment]
lgtm, thanks!

**Comment by smarterclayton** [comment]
@pweil- can you sign off?

**Comment by pweil-** [comment]
LGTM

**Comment by smarterclayton** [comment]
[merge]

**Comment by liggitt** [comment]
`'[ "$(oc rsh ${BUILD_ID}-build 2>&1 | grep 'forbidden')" ]' exited with status 1` flake fixed, re[test][merge]

**Pull request merged**